src/Model/GServer.php

```diff
@@ -2048,19 +2051,22 @@ public static function setProtocol(int $gsid, int $protocol)
 	 *
 	 * @param int $gsid Server id
 	 * @return int
-	 * @throws Exception
+	 * @throws InvalidArgumentException If $gsid is not valid
+	 * @throws ResultException If $gsid cannot be found in table
 	 */
 	public static function getProtocol(int $gsid): int
 	{
-		if (empty($gsid)) {
-			return null;
+		if ($gsid < 1) {
+			throw new InvalidArgumentException('gsid=' . $gsid . ' is invalid');
 		}
 
 		$gserver = DBA::selectFirst('gserver', ['protocol'], ['id' => $gsid]);
-		if (DBA::isResult($gserver)) {
-			return $gserver['protocol'];
+		if (!DBA::isResult($gserver)) {
+			// No record found
+			throw new ResultException('gsid=' . $gsid . ' does not have a record.');
```

Comment:
If you want to throw exceptions, then you have to catch them if it isn't critical. In this case, it isn't critical to get any protocol from this method, so you should catch it anywhere it's called.

Reply:
Why care about runtime exceptions, you might want to ask:
That's why languages like Java differentiates between checked and unchanged exceptions. An unchecked exception doesn't need to be checked, means caught (no `catch` block in method body or `throws` keyword needed in method signature). They all inherit from the `java.lang.RuntimeException` exception and the Java compiler will never complain about it when you don't catch/throw them. But you are surely allowed to do so!

Checked exceptions however inherit from `java.lang.Exception` must be a) caught e.g. explicit (best) or implicit (**don't** just catch `java.lang.Exception`!!!) or b) being thrown again by the method definition by a `throws` keyword, simply because you don't want to take care of that exception in that method and an other one should do so.

So for a normal case, these runtime-exceptions aren't critical (when everything work as expected, no logic error, ...). But in rare cases these "non-critical" exceptions can be helpful and sometimes even must come. The most famous exception in Java here is `java.lang.NullPointerException`. I might mean you missed to call a setter. Surely you can let the JVM throw that exception at you but sometimes it is then harder to find the cause because your code (and backtrace) has hundreds of thousands of classes and methods.

There are tons of examples out there for checked versus unchecked exceptions under Java (and yes, I know, I got a bit dragged away here ;-)). So what does this mean for Friendica as it is written in PHP? We can:

- InvalidArgumentException should NEVER appear but needs to be thrown when an invalid argument e.g. empty network name (empt string) is being handled over to a method
- This means that some pre-checks have failed on an other position, e.g. in the controller and caused this

So for now, if you still don't understand, understand me that I don't wish to give you here a training course in OOP and MVC pattern. I would write it differently, methods are methods, not procedures. They answer with exceptions not `NULL` or `FALSE` or anything else (which are values, not exceptions) when an other method has invoked it incorrectly (because of logic error, missing pre-checks).

Image a method `Math::divide(number $a, number $b)` and you handle a zero $b over there.

Some fake PHP code:
````
class Math
{
	// "number" is a fake scalar for any numeric values
	public static function divide (number $a, number $b): number
	{
		if (is_zero($b)) {
			throw new DivisionByZeroException('Parameter "b" is zero');
		}
		
		return $a / $b;
	}
}
````
If you leave out the check on the parameter `$b` here, and error on case `$b = 0` is getting ugly returned by the PHP engine and not nicely (and unit-testable) by an exception. `DivisionByZeroException` is inheriting from `RuntimeException` and is unchecked and doesn't need to be checked (means caught or thrown on) as it happens on run-time (I assume a fake fact again that PHP compiles). `is_zero(number $number): bool` is a fake function that checks if they (decimals, floats, single/double) are zero or not.

Just think about it for a moment how you can test that above method on its proper behavior with or without the zero check ...

Reply:
Thank you for this although I don't fully get how it relates to my comment. I'm not against exceptions but in PHP, if they aren't caught in your code, it interrupts the whole script. Sometimes it is warranted (the `TypeError` exceptions you've received during your refactoring PRs), but in this case it's not. The current code is handling missing `gserver` records gracefully and any code that refactors it should behave the same way, exceptions or not.